### DISCUSSION TITLE
Store Enum as DBObject equivalent

### DISCUSSION THREAD

**Discussion by arivazhagan-jeganathan** [MorphiaOrg/morphia#1997] [Q&A]
In Morphia 1.5.3, used toDBObject method to store Enum as an DBObject.

in Morphia 2.2.3 need to retain same functionality.. Is it possible to store Enum to same format as earlier? currently it is getting stored as String always.

Any suggestions please @evanchooly. Thanks in advance!

**Comment by evanchooly** [comment]
There isn't currently a way to do that.  A vague memory of a configuration that allowed that is dancing at the edges of perception but I can find no trace of it.  Probably your best bet to get you going today is to provide a custom codec for that enum type and register that with the driver.  If you tweak what Morphia is [using](https://github.com/MorphiaOrg/morphia/blob/d6a5567adb8d1d44d045d745bb0ea967932b8ea8/core/src/main/java/dev/morphia/mapping/codec/EnumCodec.java#L16), it should be relatively straight forward.  You can find an example of that in the [driver documentation](https://www.mongodb.com/docs/drivers/java/sync/current/fundamentals/data-formats/codecs/#std-label-codecs-custom-example).

**Reply by burdeynyy** [reply]
I am using Morphia version 2.2.3 and trying to use a custom codec for an Enum. Since my codec which I registered in the driver is not in the first candidates, EnumCodecProvider is invoked earlier and`EnumCodec` is used then. How do I add a codec which will be used by Morphia? 
P.S. I just need an enum codec which will be ignoring unknown values and returning null in this case.
